### PR TITLE
Display admin order page instead of shopfront order page to avoid error 500

### LINKED ISSUE #12771: [Subscriptions] Error 500 when clicking link to order of a failed subscription due to stock zero
## Description

If all products of a subscription order have stock 0, then the 'summary of recently placed subscription orders' email (which is sent to the coordinator) contains a link to the order in the shopfront, e.g. `/orders/Rxxx`. However, clicking the link shows an error 500. I believe this is, because the order wasn't placed at all (it is still in 'cart' state) and at that stage that page doesn't even exist. 

Bugsnag: https://app.bugsnag.com/yaycode/open-food-network-germany/errors/661933e3c27d120009d590e3?filters[error.status]=open&filters[event.since]=30d

## Expected Behavior

There should be no error 500.
The link should point to a valid page, e.g. the order's page in the admin section (which does exist):  `/admin/orders/Rxxx/edit`.
Bonus: It would be very helpful to find information about which products had zero stock, how many had been tried to be ordered and who the customer was (email).

## Actual Behaviour

There's an error 500 and no further information available.

## Steps to Reproduce

1. Set up a subscription order for a product which has stock 0.
2. Open the order cycle, so the subscription order gets placed.
3. Check the 'summary of recently placed subscription orders' email.
4. Click the link to order.

## Workaround

Find the order in the orders list in the admin area. However this is not obvious, because the checkbox to only show complete orders has to unchecked.

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by drummer83**
#### What? Why?

- Closes #12771  

The 'summary of recently placed subscription orders' email contained a broken link in case all product of an order are out of stock (stock = 0), which causes an error 500.
image

The linked /orders/Rxxx page cannot be used while an order is still in state 'cart'. The error 500 is triggered because the shipping method has not defined at this stage yet. 
image

This PR addresses this problem by linking to the order's page in the admin area instead. This page exists as soon as the order is created, also in state 'cart'. I think that this page is the better place to link to anyways, because it provides all information and options to edit everything for the enterprise user (who is receiving the email).
image

#### What should we test?

- Make a subscription order fail for different reasons, e.g. No stock at all, insufficient stock, failed payment, already processed etc.
- Check the 'summary of recently placed subscription orders' email.
- Click the link to the order mentioned in the email.
- Verify that the edit order page for that order is shown.
- Verify that the emails sent to customers remain unchanged.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [x] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [ ] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: app/views/subscription_mailer/_summary_detail.html.haml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
   - separator = messages.values.any? ? ": " : ", "
   - orders.each_with_index do |order, i|
-    %a{ href: order_url(order) }>= order.number
+    %a{ href: spree.edit_admin_order_url(order) }>= order.number
```
Comment (mkllnk) [inline app/views/subscription_mailer/_summary_detail.html.haml:9]: Just for context, this partial is used in placement_summary_email and confirmation_summary_email which are both sent to the shop's contact email.

I just wanted to check that none of this is sent to customers. It looks fine.

**Review by mkllnk — APPROVED** [review]

File: app/views/subscription_mailer/_summary_detail.html.haml (hunk 1 above)
Comment (dacook) [reply app/views/subscription_mailer/_summary_detail.html.haml:9]: Thanks for checking this. Perhaps in the future we should arrange mailer views according to their audience, with an `admin/` folder for those that are sent to admin users.

**Review by dacook — APPROVED** [review]
Well done on both documenting this, and fixing it!

**Comment by drummer83** [comment]
@RachL @filipefurtad0 Could one of you two please test this one as I'm the author?
Thanks!

**Comment by filipefurtad0** [comment]
Hey @drummer83 ,

Nice! Thanks for this PR.

I could reproduce the bug with a stock restriction test case, and observe that after staging this PR:
- the link on the 'summary of recently placed subscription orders' email points to the respective order edit page. The order is still on cart state, as expected.
- verified that this PR introduces no changes to customers emails.

Awesome :muscle: 
Good to go.

**Pull request merged**